unstructured_ingest/v2/processes/connectors/github.py

```
@@ -0,0 +1,264 @@
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                date_created = datetime.strptime(
+                    element._headers["date"],
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                yield FileData(
+                    identifier=element.sha,
+                    connector_type=CONNECTOR_TYPE,
+                    source_identifiers=SourceIdentifiers(
+                        fullpath=element.path,
+                        filename=element.path.split("/")[-1],
+                        rel_path=rel_path,
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element.url,
+                        version=element.etag,
+                        record_locator=record_locator,
+                        date_modified=date_modified,
+                        date_created=date_created,
+                        date_processed=str(time()),
+                    ),
+                    additional_metadata={
+                        "content-type": element._headers.get("content-type", ""),
+                        "mode": element._rawData.get("mode", ""),
+                        "type": element._rawData.get("type", ""),
+                        "size": element._rawData.get("size", ""),
+                    },
+                )
+
+
+class GitHubDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitHubDownloader(Downloader):
+    connection_config: GitHubConnectionConfig
+    download_config: GitHubDownloaderConfig
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_file(self, path):
+        from github.GithubException import UnknownObjectException
+
+        try:
+            content_file = self.connection_config.get_repo().get_contents(path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exists {self.connection_config.url}/{path}")
+            return None
```

Let's reraise this as an error rather than returning None, I don't see any reason why we this error should be thrown without us being notified of it.